### ISSUE TITLE
Deployment timeout

### ISSUE THREAD

**Issue by rgarrigue** [biemond/biemond-wildfly#167]
Hello

Just for an advice. Here's the deployment debug

``` 
==> testcentos.poc: Notice: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/File[/tmp/xwiki-enterprise-web-8.4.3.war]/owner: owner changed 'root' to 'wf10'
==> testcentos.poc: Notice: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/File[/tmp/xwiki-enterprise-web-8.4.3.war]/group: group changed 'root' to 'wf10'
==> testcentos.poc: Notice: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/File[/tmp/xwiki-enterprise-web-8.4.3.war]/mode: mode changed '0644' to '0655'
==> testcentos.poc: Debug: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/File[/tmp/xwiki-enterprise-web-8.4.3.war]: The container Wildfly::Deployment[xwiki.war] will propagate my refresh event
==> testcentos.poc: Debug: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/File[/tmp/xwiki-enterprise-web-8.4.3.war]: The container Wildfly::Deployment[xwiki.war] will propagate my refresh event
==> testcentos.poc: Debug: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/File[/tmp/xwiki-enterprise-web-8.4.3.war]: The container Wildfly::Deployment[xwiki.war] will propagate my refresh event
==> testcentos.poc: Debug: Wildfly_deployment[xwiki.war](provider=http_api): Exists? xwiki.war
==> testcentos.poc: Debug: Wildfly_deployment[xwiki.war](provider=http_api): Deploying xwiki.war from source /tmp/xwiki-enterprise-web-8.4.3.war
==> testcentos.poc: Error: Could not set 'present' on ensure: Net::ReadTimeout at 51:/etc/puppet/environments/production/modules/wildfly/manifests/deployment.pp
==> testcentos.poc: Error: Could not set 'present' on ensure: Net::ReadTimeout at 51:/etc/puppet/environments/production/modules/wildfly/manifests/deployment.pp
==> testcentos.poc: Wrapped exception:
==> testcentos.poc: Net::ReadTimeout
==> testcentos.poc: Error: /Stage[main]/Profiles::Wildfly/Wildfly::Deployment[xwiki.war]/Wildfly_deployment[xwiki.war]/ensure: change from absent to present failed: Could not set 'present' on ensure: Net::ReadTimeout at 51:/etc/puppet/environments/production/modules/wildfly/manifests/deployment.pp
```

Do you know which timeout this is ? I've upped the timeout value to 15 min, I'm far from it. And Jolokia's ok, so smaller war do fine but xwiki's HS.

```
profiles::wildfly::deployments:
  'jolokia.war':
    source: 'http://foreman01.prov:8080/public/war/jolokia-war-1.3.5.war'
  'xwiki.war':
    source: 'http://foreman01.prov:8080/public/war/xwiki-enterprise-web-8.4.3.war'
    timeout: 900
    operation_headers:
      blocking-timeout: 900
```

Thanks in advance !

**Comment by jairojunior** [comment]
Sorry. It's a bug.

`wildfly_deployment` should be passing timeout ahead here: https://github.com/biemond/biemond-wildfly/blob/v1.0.1/lib/puppet/provider/wildfly_deployment/http_api.rb#L9

But it's not. :frowning_face: 

It's super easy to fix, but I'm in the middle of 1.1.0 release. If you're in a hush just add `resource[:timeout]` as an argument to APIClient constructor.

**Comment by rgarrigue** [comment]
Ah ok.

Well, I fixed this with more memory for my VM, which was dying from swap anyway, the deployment went bellow the threshold. 

Fix this whenever you can, no hurry for me. Just, should I use only timeout, using both timeout and  operation_headers / blocking timeout ? 

Thanks for your reactivity

**Comment by jairojunior** [comment]
They're different things, `wildfly_deployment``s timeout is used in Management API interactions because a deployment operation is an upload operation + actual deployment and both takes some time, the first depends on the size of package and second on how many subsystems that application loads, since Wildfly have a micro-kernel architecture whereas subsystems are lazily loaded.

To understand `blocking-timeout` read this: https://docs.jboss.org/author/display/WFLY9/Admin+Guide#AdminGuide-Controllingoperationblockingtime

**TL;DR;** They ideally have the same value.

Just fixed `wildfly_deployment`, but as I wrote this I realized that timeout default value should be increased to `300` to match `blocking-timeout`. Will fix this too.
